library/src/main/java/com/otaliastudios/zoom/ZoomEngine.kt

```diff
@@ -625,10 +637,25 @@ internal constructor(context: Context) : ViewTreeObserver.OnGlobalLayoutListener
         return value.coerceIn(minZoom, maxZoom)
     }
 
+    /**
+     * Uses [checkPanBounds] to calculate the pan correction that
+     * needs to be applied to get into a valid state.
+     *
+     * @return the scaled pan fix values for x- and y-axis
+     */
+    private fun getPanCorrection(): ScaledPoint {
+        return ScaledPoint(
```

Comment:
See if it's better to avoid this new instance as well depending on how frequently this is accessed